src/ol/format/ogc/filter.js

```diff
@@ -376,6 +408,82 @@ ol.format.ogc.filter.Bbox = function(geometryName, extent, opt_srsName) {
 ol.inherits(ol.format.ogc.filter.Bbox, ol.format.ogc.filter.Filter);
 
 
+/**
+ * @classdesc
+ * Represents a `<Intersects>` operator to test whether a geometry-valued property
+ * intersects a fixed bounding geometry
+ *
+ * @constructor
+ * @param {!string} geometryName Geometry name to use.
+ * @param {!ol.geom.Geometry} geometry Geometry.
+ * @param {string=} opt_srsName SRS name. No srsName attribute will be
+ *    set on geometries when this is not provided.
+ * @extends {ol.format.ogc.filter.Filter}
+ * @api
+ */
+ol.format.ogc.filter.Intersects = function(geometryName, geometry, opt_srsName) {
+
+  ol.format.ogc.filter.Filter.call(this, 'Intersects');
+
+  /**
+   * @public
+   * @type {!string}
+   */
+  this.geometryName = geometryName || 'the_geom';
+
+  /**
+   * @public
+   * @type {ol.geom.Geometry}
+   */
+  this.geometry = geometry;
+
+  /**
+   * @public
+   * @type {string|undefined}
+   */
+  this.srsName = opt_srsName;
+};
+ol.inherits(ol.format.ogc.filter.Intersects, ol.format.ogc.filter.Filter);
+
+
+/**
+ * @classdesc
+ * Represents a `<Within>` operator to test whether a geometry-valued property
+ * within a fixed bounding geometry
+ *
+ * @constructor
+ * @param {!string} geometryName Geometry name to use.
+ * @param {!ol.geom.Geometry} geometry Geometry.
+ * @param {string=} opt_srsName SRS name. No srsName attribute will be
+ *    set on geometries when this is not provided.
+ * @extends {ol.format.ogc.filter.Filter}
+ * @api
+ */
+ol.format.ogc.filter.Within = function(geometryName, geometry, opt_srsName) {
```

Comment:
Class `ol.format.ogc.filter.Within` and `ol.format.ogc.filter.Intersects` are basically the same. Could you refactor it a bit to avoid the duplicated code and to make it easier to add new spatial filters? Something like:

https://github.com/tsauerwein/ol3/commit/81abc3b078b92f94e951e94c9cd3a27574fdc59c